File: webapp/src/app/models/final-grade.ts
import { Convert } from './index';
import { fromJsonToUser, User, UserJson } from './user.model';

export interface FinalGradeJson {
  score?: string;
  user?: UserJson;
}

export interface FinalGrade {
  score?: string;
  user?: User;
}

export const fromJsonToFinalGrade: Convert<FinalGradeJson, FinalGrade> = (json: FinalGradeJson) => {
  return (!json) ? undefined : {
    ...json,
    user: fromJsonToUser(json?.user)
  };
}

File: api/src/main/java/br/niedunicamp/pojo/FinalGrade.java
package br.niedunicamp.pojo;

import br.niedunicamp.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FinalGrade {
    private String score;
    private User user;
}